lib/chat_web/live/main_live/index.ex

```diff
@@ -390,56 +412,170 @@ defmodule ChatWeb.MainLive.Index do
     )
   end
 
-  defp allow_chunked_upload(socket, type, opts) do
+  defp allow_file_upload(socket) do
     socket
-    |> allow_upload(type,
-      auto_upload: true,
-      max_file_size: 102_400_000_000,
+    |> allow_upload(:file,
       accept: :any,
-      max_entries: Keyword.get(opts, :max_entries, 1),
+      auto_upload: true,
       external: &chunked_presign_url/2,
+      max_entries: 2000,
+      max_file_size: 102_400_000_000,
       progress: &handle_chunked_progress/3
     )
+    |> assign(:uploads_metadata, %{})
   end
 
   def chunked_presign_url(entry, socket) do
-    {key, secret} = ChunkedFiles.new_upload()
-    socket = start_chunked_upload(socket, entry, key, secret)
+    upload_key = get_upload_key(entry, socket)
+
+    {next_chunk, key, secret} =
+      case UploadIndex.get(upload_key) do
+        nil ->
+          {key, secret} = ChunkedFiles.new_upload()
```

Comment:
Since `upload_key` should be unique. We might use it as file_chunk key.
This line becomes `secret = ChunkedFiles.new_upload(upload_key)`.
And there would be no need in `key` as we have `upload_key` instead.

This also improves upcoming feature - building partial DB dump based on user scope.

Reply:
That was my first approach, but we use the chunk_key in the /upload_chunk/:key URL, so I'm worried there would be issues with the URL limit.

Reply:
now it is hashed so  will fit into URL